src/components/CopyStateLinkButton/index.tsx

```diff
@@ -0,0 +1,139 @@
+import { RefObject, useEffect, useState } from 'react';
+import DoneIcon from '@mui/icons-material/Done';
+import LinkIcon from '@mui/icons-material/Link';
+import { useAppDispatch, useAppSelector } from 'redux/hooks';
+import { selectChartDatePeriod } from 'redux/ChartView/selectors';
+import {
+    selectCurrentDate,
+    selectSelectedCountryInSideBar,
+    selectTimeseriesDates,
+} from 'redux/App/selectors';
+import { URLToFilters } from 'utils/helperFunctions';
+import { setPopup, setSelectedCountryInSidebar } from 'redux/App/slice';
+import { selectCountriesData } from '../../redux/App/selectors';
+import Snackbar from '@mui/material/Snackbar';
+import Alert from '@mui/material/Alert';
+import { CopyStateLinkButtonContainer } from './styled';
+
+interface CopyStateLinkButtonProps {
+    onWhichContainer: 'view' | 'chart';
+    map?: RefObject<mapboxgl.Map | null>;
+}
+
+const CopyStateLinkButton = ({
+    onWhichContainer,
+    map,
+}: CopyStateLinkButtonProps) => {
+    const dispatch = useAppDispatch();
+
+    const chartDatePeriod = useAppSelector(selectChartDatePeriod);
+    const selectedCountry = useAppSelector(selectSelectedCountryInSideBar);
+    const countriesData = useAppSelector(selectCountriesData);
+    const timeseriesDates = useAppSelector(selectTimeseriesDates);
+    const currentDate = useAppSelector(selectCurrentDate);
+
+    useEffect(() => {
+        const newViewValues = URLToFilters(location.search);
+
+        if (!newViewValues.name || countriesData.length === 0) return;
+
+        if (map && map.current) {
+            const mapRef = map.current;
+            mapRef.setCenter([newViewValues.lng || 40, newViewValues.lat || 0]);
+            mapRef.setZoom(newViewValues.zoom || 2.5);
+        }
+        for (const country of countriesData) {
```

Comment:
Here you can use `find` function on the `countriesData` array. It would simplify code a little bit